test/integration/main_test.go

```diff
@@ -0,0 +1,416 @@
+- gcloud projects list --format json --uri --filter projectId:INTEG_TEST_PROJECT_ID
+- gcloud projects list --uri --filter projectId:INTEG_TEST_PROJECT_ID
+`, "INTEG_TEST_PROJECT_ID", cfg.ProjectID),
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			args := []string{
+				"tool", "cleanup",
+				"-path", tc.filePath,
+			}
+			if tc.verbose {
+				args = append(args, "-verbose")
+			}
+
+			_, stdout, stderr, err := cli.PipeAndRun(ctx, args)
+			if err != nil {
+				t.Fatalf("Process(%+v) failed to run command %v", tc.name, err)
+			}
+
+			if strings.TrimSpace(tc.wantOutput) != strings.TrimSpace(stdout.String()) {
+				t.Errorf("Process(%+v) output response got %q, want %q)", tc.name, stdout.String(), tc.wantOutput)
+			}
+			if stderr.String() != "" {
+				t.Errorf("Process(%+v) got unexpected error: %q)", tc.name, stderr.String())
+			}
+		})
+	}
+}
+
+func TestToolValidate(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name       string
+		filePath   string
+		wantOutput string
+	}{
+		{
+			name:       "success",
+			filePath:   fmt.Sprintf("../../%s/tool.yaml", cfg.WorkingDir),
+			wantOutput: "Successfully validated tool request",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			args := []string{
+				"tool", "validate",
+				"-path", tc.filePath,
+			}
+
+			_, stdout, stderr, err := cli.PipeAndRun(ctx, args)
+			if err != nil {
+				t.Fatalf("Process(%+v) failed to run command %v", tc.name, err)
+			}
+
+			if strings.TrimSpace(tc.wantOutput) != strings.TrimSpace(stdout.String()) {
+				t.Errorf("Process(%+v) output response got %q, want %q)", tc.name, stdout.String(), tc.wantOutput)
+			}
+			if stderr.String() != "" {
+				t.Errorf("Process(%+v) got unexpected error: %q)", tc.name, stderr.String())
+			}
+		})
+	}
+}
+
+func testGetPolicyBindings(ctx context.Context, tb testing.TB, cfg *config) ([]*iampb.Binding, error) {
+	tb.Helper()
+
+	req := &iampb.GetIamPolicyRequest{
+		Resource: fmt.Sprintf("projects/%s", cfg.ProjectID),
+		Options: &iampb.GetPolicyOptions{
+			RequestedPolicyVersion: 3,
+		},
+	}
+	backoff := retry.WithMaxRetries(cfg.QueryRetryLimit, retry.NewConstant(cfg.QueryRetryWaitDuration))
+	var bs []*iampb.Binding
+
+	if err := retry.Do(ctx, backoff, func(ctx context.Context) error {
+		p, err := projectClient.GetIamPolicy(ctx, req)
+		if err != nil {
+			return retry.RetryableError(fmt.Errorf("failed to get IAM policy: %w", err))
+		}
+		bs = p.Bindings
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return bs, nil
+}
+
+func testGotBindings(tb testing.TB, bs []*iampb.Binding, matchTitle string) (result []*iampb.Binding) {
+	tb.Helper()
+
+	for _, b := range bs {
+		if b.Condition != nil && b.Condition.Title == matchTitle {
+			result = append(result, b)
```

Comment:
Do we need to ensure the order of the bindings to compare effectively?

Reply:
We already checks the order by using list?

Reply:
I can change the order of the bindings to descending in the request file, and check if the want bindings are in ascending order.

Reply:
Wait, actually we don't guarantee the order IAM bindings, we only guarantee the order or IAM binding members.